www/src/components/shell/onboarding/sections/repository/ProviderConfiguration.tsx

```diff
@@ -0,0 +1,191 @@
+import {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+} from 'react'
+import { useNavigate } from 'react-router-dom'
+import { Flex, Img, Input } from 'honorable'
+
+import {
+  Button,
+  FormField,
+  ListBoxItem,
+  Select,
+} from '@pluralsh/design-system'
+
+import { OnboardingContext } from '../../context/onboarding'
+import { ScmProvider } from '../../../../../generated/graphql'
+import { isAlphanumeric } from '../../../helpers'
+import { OrgType, SCMOrg } from '../../context/types'
+
+import { useGithubState } from './provider/github'
+import { useGitlabState } from './provider/gitlab'
+
+interface OrgInputProps {
+  orgs: Array<SCMOrg>
+}
+
+function OrgInput({ orgs }: OrgInputProps) {
+  const { scm: { provider, org }, setSCM } = useContext(OnboardingContext)
+  const setOrg = useCallback(org => setSCM(scm => ({ ...scm, org })), [setSCM])
+  const setOrgByKey = useCallback(id => {
+    const org = orgs.find(o => o.id === id)
+
+    setSCM(scm => ({ ...scm, org }))
+  }, [orgs, setSCM])
+
+  useEffect(() => {
+    if (org) return
+
+    const userOrg = orgs.find(o => o.orgType === OrgType.User)
+
+    if (!userOrg) return
+
+    setOrg(userOrg)
+  }, [org, orgs, setOrg])
+
+  return (
+    <FormField
+      width="100%"
+      label={
+        provider === ScmProvider.Github
+          ? 'GitHub organization'
+          : provider === ScmProvider.Gitlab
+            ? 'GitLab group'
+            : 'Organization or group'
```

Comment:
Can you add brackets to the nested if?

Reply:
Refactored that to a function to increase readability.